build-tools/packages/build-cli/docs/generate.md

```diff
@@ -105,3 +106,60 @@ DESCRIPTION
 
   Customize the code URL prefix by setting oclif.repositoryPrefix in package.json.
 ```
+
+## `flub generate typetests`
+
+Generates type tests based on the individual package settings in package.json.
+
+```
+USAGE
+  $ flub generate typetests [-d <value> | --packages | -g client|server|azure|build-tools] [--prepare | --generate]
+    [--exact <value> |  | [-s
+    ^previousMajor|^previousMinor|~previousMajor|~previousMinor|previousMajor|previousMinor|baseMinor|baseMajor | ]]
+    [--reset | ] [-v]
+
+FLAGS
+  -d, --dir=<value>                 Run on the package in this directory.
+  -g, --releaseGroup=<option>       Run on all packages within this release group.
+                                    <options: client|server|azure|build-tools>
+  -s, --versionConstraint=<option>  [default: baseMinor] The type of version constraint to use for previous versions.
+                                    Only applies to the prepare phase.
+                                    <options: ^previousMajor|^previousMinor|~previousMajor|~previousMinor|previousMajor|
+                                    previousMinor|baseMinor|baseMajor>
+  -v, --verbose                     Verbose logging.
+  --exact=<value>                   An exact string to use as the previous version constraint. The string will be used
+                                    as-is. Only applies to the prepare phase.
+  --generate                        Generates tests only. Doesn't prepare the package.json.
+  --packages                        Run on all independent packages in the repo.
+  --prepare                         Prepares the package.json only. Doesn't generate tests. Note that npm install may
+                                    need to be run after preparation.
+  --reset                           Resets the broken type test settings in package.json. Only applies to the prepare
+                                    phase.
+
+DESCRIPTION
+  Generates type tests based on the individual package settings in package.json.
+
+  Generating type tests has two parts: preparing package.json and generating type tests. By default, both steps are run
+  for each package. This can be overridden using the --prepare and --generate flags.
+
+EXAMPLES
+  Prepare the package.json for all packages in the client release group.
```

Comment:
What's the diff between "prepare" and "pin" in this case? Using -g option without -s ?

Reply:
Sorry I don't quite understand the question, so this may not answer it. :) "Prepare" is used to set the previous version in package.json to the right version based on the constraint passed in (`-s`). `-s` has a default of "`baseMinor` -- I could remove the default to make it explicit what type of constraint you're using. What do you think?

Anyway, "pinning" is when you specify either `baseMinor`, `baseMajor`, `previousMinor`, or `previousMajor`. That will "pin" the previous version to that specific baseline. The other `-s` values set a range instead of a "pinned" version. Maybe I should avoid the "pinning" term altogether?

BTW, I changed the `-g` flag to be the release group flag, because that's what we use for all the other commands. I figure since so few folks know about the type tests anyway, this change in the "interface" was OK. Let me know if you disagree.

Reply:
You navigated the answer well given you did not understand the question :) Maybe this language around "examples" confused me. I'm clear now.

Yes - given that (a) typetests are so tightly correlated to versions, and (b) it's hard to reason what "default" you could expect, I'd make -s required option.